app/models/feedback_message.rb

```diff
@@ -8,18 +8,24 @@ class FeedbackMessage < ApplicationRecord
   has_one :email_message, dependent: :nullify
   has_many :notes, as: :noteable, inverse_of: :noteable, dependent: :destroy
 
+  REPORTER_UNIQUENESS_SCOPE = %i[reported_url feedback_type].freeze
+  REPORTER_UNIQUENESS_MSG = "(you) previously reported this URL.".freeze
+  CATEGORIES = ["spam", "other", "rude or vulgar", "harassment", "bug", "listings"].freeze
```

Comment:
It kind of makes you wonder why this and the statuses aren't Rails `enum`s 🤔 This would give us validations for free and also scopes, etc. 

Somewhat contrary to popular belief, Rails `enum`s can also work with string columns, but it means you need to use the hash form for defining them:

I did the following locally and it works well:

```diff
diff --git a/app/models/feedback_message.rb b/app/models/feedback_message.rb
index 2527fedda..a4cee1bfc 100644
--- a/app/models/feedback_message.rb
+++ b/app/models/feedback_message.rb
@@ -10,20 +10,20 @@ class FeedbackMessage < ApplicationRecord

   REPORTER_UNIQUENESS_SCOPE = %i[reported_url feedback_type].freeze
   REPORTER_UNIQUENESS_MSG = "(you) previously reported this URL.".freeze
-  CATEGORIES = ["spam", "other", "rude or vulgar", "harassment", "bug", "listings"].freeze
-  STATUSES = %w[Open Invalid Resolved].freeze
+  enum category: {
+    spam: "spam",
+    other: "other",
+    rude: "rude or vulgar",
+    harassment: "harassment",
+    bug: "bug",
+    listings: "listings"
+  }
+
+  enum status: { open: "Open", invalid: "Invalid", resolved: "Resolved" }, _prefix: :status

   validates :feedback_type, :message, presence: true
   validates :reported_url, :category, presence: { if: :abuse_report? }, length: { maximum: 250 }
   validates :message, length: { maximum: 2500 }
-  validates :category,
-            inclusion: {
-              in: CATEGORIES
-            }
-  validates :status,
-            inclusion: {
-              in: STATUSES
-            }
```

Note that I specified `_prefix: :status` for the second enum because it contains `invalid` as an option which conflicts with an AR method. Usage:

```ruby
# scope
FeedbackMessage.status_open.count
#=>   (1.0ms)  SELECT COUNT(*) FROM "feedback_messages" WHERE "feedback_messages"."status" = $1  [["status", "Open"]]
=> 2

# validation
feedback_message.status = "test"
ArgumentError: 'test' is not a valid status
```

Reply:
Meant to ask, but do you think I should switch it to an enum? I think I probably shouldn't because we're relying on catching the validation message appropriately and I think a raise of `ArgumentError` would cause the response to 500.

Reply:
No, you probably shouldn't, at least not as part of this PR.